lib/src/cpp/domain_participant/DomainParticipant.cpp

```diff
@@ -20,314 +20,437 @@
+        return;
+    }
+
+    // Try to set up Xerces tools
+    if (!initialize_tools(config, doc, error_handler, implementation, output, parser, serializer, target, absolute_xml_file))
+    {
+        // Close XML workspace
+        xercesc::XMLPlatformUtils::Terminate();
+        return;
+    }
+
+    // Obtain profiles node
+    try
+    {
+        profiles_node = get_node(doc, eprosima::qosprof::tag::PROFILES);
+    }
+    catch (const eprosima::qosprof::ElementNotFound& ex)
+    {
+        // Obtain root element
+        xercesc::DOMElement* root_element = doc->getDocumentElement();
+
+        // Add profiles
+        profiles_node = (xercesc::DOMNode*) doc->createElement(xercesc::XMLString::transcode(eprosima::qosprof::tag::PROFILES));
+        root_element->appendChild(profiles_node);
+    }
+
+    // Obtain participant node with the profile id
+   try
+    {
+        participant_node = get_node(
+            profiles_node,
+            eprosima::qosprof::tag::PARTICIPANT,
+            eprosima::qosprof::tag::PROFILE_NAME,
+            profile_id);
+    }
+    catch (const eprosima::qosprof::ElementNotFound& ex)
+    {
+        // Create if not exist
+        xercesc::DOMElement * participant_element = doc->createElement(
+            xercesc::XMLString::transcode(eprosima::qosprof::tag::PARTICIPANT));
+        profiles_node->appendChild(participant_element);
+        participant_element->setAttribute(
+            xercesc::XMLString::transcode(eprosima::qosprof::tag::PROFILE_NAME),
+            xercesc::XMLString::transcode(profile_id.c_str()));
+        participant_node = (xercesc::DOMNode*)participant_element;
+
+    }
+
+    // Obtain rtps node
+    try
+    {
+        rtps_node = get_node(participant_node, eprosima::qosprof::tag::RTPS);
+    }
+    catch (const eprosima::qosprof::ElementNotFound& ex)
+    {
+        // Create if not exist
+        rtps_node = (xercesc::DOMNode*) doc->createElement(xercesc::XMLString::transcode(
+            eprosima::qosprof::tag::RTPS));
+        participant_node->appendChild(rtps_node);
+
+    }
+
+    // Obtain name
+    try
+    {
+        name_node = get_node(rtps_node, eprosima::qosprof::tag::NAME);
+    }
+    catch (const eprosima::qosprof::ElementNotFound& ex)
+    {
+        // Create if not exist
+        name_node = (xercesc::DOMNode*) doc->createElement(xercesc::XMLString::transcode(
+            eprosima::qosprof::tag::NAME));
+        rtps_node->appendChild(name_node);
+    }
+
+    // Remove all childs
+    reset_node(name_node);
+
+    // Set name value
+    name_value = doc->createTextNode(xercesc::XMLString::transcode(name.c_str()));
+    name_node->appendChild(name_value);
+
+    // Validate new XML element and save it
+    validate_and_save_xml_document(doc, error_handler, output, parser, serializer, target);
+
+    // Close XML workspace
+    xercesc::XMLPlatformUtils::Terminate();
+    return;
```

Comment:
I do not think we need this return being a void function and leaving the function scope just below.